src/script/lua_api/l_object.cpp

```diff
@@ -433,21 +453,64 @@ int ObjectRef::l_get_physics_override(lua_State *L)
 		return 0;
 	// Do it
 	lua_newtable(L);
-	lua_pushnumber(L, co->m_physics_override_speed);
+	lua_pushnumber(L, co->m_physics_override.speed);
 	lua_setfield(L, -2, "speed");
-	lua_pushnumber(L, co->m_physics_override_jump);
+	lua_pushnumber(L, co->m_physics_override.jump);
 	lua_setfield(L, -2, "jump");
-	lua_pushnumber(L, co->m_physics_override_gravity);
+	lua_pushnumber(L, co->m_physics_override.gravity);
 	lua_setfield(L, -2, "gravity");
-	lua_pushboolean(L, co->m_physics_override_sneak);
+	lua_pushboolean(L, co->m_physics_override.sneak);
 	lua_setfield(L, -2, "sneak");
-	lua_pushboolean(L, co->m_physics_override_sneak_glitch);
+	lua_pushboolean(L, co->m_physics_override.sneak_glitch);
 	lua_setfield(L, -2, "sneak_glitch");
 	lua_pushboolean(L, co->m_physics_override_new_move);
 	lua_setfield(L, -2, "new_move");
 	return 1;
 }
 
+// set_physics_modifier(self, key, modifier_table)
+int ObjectRef::l_set_physics_modifier(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	const std::string key = readParam<std::string>(L, 2);
+	PlayerSAO::PhysicsModifier modifier;
+	if (!read_physics_modifier(L, 3, modifier))
+		throw LuaError("set_physics_modifier expects (string, table)");
+
+	if (PlayerSAO *player = dynamic_cast<PlayerSAO*>(getobject(ref))) {
+		player->setPhysicsModifier(key, modifier);
+		
+		if (player->m_physics_override_set) {
+			warningstream << "set_physics_modifier will have no effect because ";
+			warningstream << "set_physics_override was previously called" << std::endl;
+			warningstream << script_get_backtrace(L);
+		}
+	}
+
+	return 0;
+}
+
+// delete_physics_modifier(self, key)
+int ObjectRef::l_delete_physics_modifier(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	const std::string key = readParam<std::string>(L, 2);
+
+	if (PlayerSAO *player = dynamic_cast<PlayerSAO*>(getobject(ref))) {
```

Comment:
In order to reduce indents, just return 0 here. It also adds safety when somebody would want to extend this function afterwards
EDIT: Same for `l_set_physics_modifier`